src/test/CMakeLists.txt

```diff
@@ -16,7 +16,6 @@ generate_header_from_raw(data/asmap.raw)
 
 add_executable(test_bitcoin
   main.cpp
-  $<TARGET_OBJECTS:bitcoin_consensus>
```

Comment:
What's this change about?

Reply:
Before CMake 3.12, the only way to consume the object library--which is a mere unarchived collection of object files, not a real library--was to add `$<TARGET_OBJECTS:bitcoin_consensus>` to sources. Now we can use an idiomatic way: `target_link_libraries(test_bitcoin bitcoin_consensus)`.